packages/opencensus-core/src/metrics/cumulative/derived-cumulative.ts

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright 2019, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getTimestampWithProcessHRTime} from '../../common/time-util';
+import {validateArrayElementsNotNull, validateNotNull} from '../../common/validations';
+import {LabelKey, LabelValue, Metric, MetricDescriptor, MetricDescriptorType, TimeSeries, Timestamp} from '../export/types';
+import {Meter} from '../types';
+import {AccessorInterface} from '../types';
+import * as util from '../utils';
+
+type ValueExtractor = () => number;
+
+interface CumulativeEntry {
+  readonly labelValues: LabelValue[];
+  readonly extractor: ValueExtractor;
+  prevValue: number;
+}
+
+/**
+ * DerivedCumulative metric is used to record aggregated metrics that
+ * represents a single numerical value accumulated over a time interval.
+ */
+export class DerivedCumulative implements Meter {
+  private metricDescriptor: MetricDescriptor;
+  private labelKeysLength: number;
+  private registeredPoints: Map<string, CumulativeEntry> = new Map();
+  private extractor?: ValueExtractor;
+  private readonly constantLabelValues: LabelValue[];
+  private startTime: Timestamp;
+
+  /**
+   * Constructs a new DerivedCumulative instance.
+   *
+   * @param {string} name The name of the metric.
```

Comment:
What would you think about removing all the JSDoc types since those are included in the TypeScript types anyway (and so feel redundant here)?

Reply:
Done, have to fix same in other places. I will open new PR to make things consistent everywhere.